os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -350,9 +353,9 @@ void binary_manager_recovery(int bin_idx)
 		bmlldbg("Failed to deactivate binary, bin idx %d\n", bin_idx);
 		goto reboot_board;
 	}
+	BIN_STATE(bin_idx) = BINARY_FAULT;
```

Comment:
How about moving this into the `binary_manager_deactivate_binary`?

Reply:
It's good. I modified.